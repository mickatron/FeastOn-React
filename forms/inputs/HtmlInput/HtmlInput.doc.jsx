/* eslint-disable */
import React from 'react';
import Component from './HtmlInput';
import PropTypes from 'prop-types';

import classNameType from '../../../proptypes/className';
/**
 * Renders an `input`.
 *
 * This component is not composed by the `uiWrapper` HOC, though it does have a similar modifier
 * and width prop.
 */
function HtmlInput(props) {
  return <Component {...props} />;
}
HtmlInput.propTypes = {
  /** A custom className */
  className: classNameType,
  /** The type of input you wish to render. Must be a valid HTML input element type. */
  type: PropTypes.oneOf([
    'checkbox',
    'email',
    'file',
    'hidden',
    'image',
    'number',
    'password',
    'radio',
    'range',
    'reset',
    'search',
    'tel',
    'text',
    'time',
    'url',
  ]),
  /** BEM modifier names to be applied to the elements wrapper. */
  modifier: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  /** Width className */
  widthClass: PropTypes.string,
}

HtmlInput.defaultProps = {
  className: undefined,
  modifier: undefined,
  type: 'text',
  width: undefined,
}
export default HtmlInput;
