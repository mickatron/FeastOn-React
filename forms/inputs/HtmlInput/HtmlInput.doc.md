### Default HtmlInput.
```js
<div>
  Text Input
  <HtmlInput 
    onChange={()=>{}}
    type="text" 
  />
  Radio Input
  <HtmlInput 
    onChange={()=>{}}
    type="radio" 
  />
  Checkbox Input
  <HtmlInput 
    onChange={()=>{}}
    type="checkbox" 
  />
  Email Input
  <HtmlInput 
    onChange={()=>{}}
    type="email" 
  />
</div>
```