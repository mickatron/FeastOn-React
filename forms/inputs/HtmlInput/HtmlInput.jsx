import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import classNameType from '../../../proptypes/className';
import { getModifiers } from '../../../proptypes/editable/modifierProps';
import { getWidths } from '../../../proptypes/editable/widthProps';

import capitalizeFirstLetter from '../../../utils/capitalizeFirstLetter';

const widths = getWidths();
const modifiersAll = getModifiers('HtmlInput', []);

/**
 * Renders an `input`.
 *
 * This component is not composed by the `uiWrapper` HOC, though it does have a similar modifier
 * and width prop.
 */
export default class HtmlInput extends React.PureComponent {
  static propTypes = {
    /** A custom className */
    className: classNameType,
    /** The type of input you wish to render. Must be a valid HTML input element type. */
    type: PropTypes.oneOf([
      'checkbox',
      'email',
      'file',
      'hidden',
      'image',
      'number',
      'password',
      'radio',
      'range',
      'reset',
      'search',
      'tel',
      'text',
      'time',
      'url',
    ]),
    /** BEM modifier names to be applied to the elements wrapper. */
    modifier: PropTypes.oneOfType([
      PropTypes.oneOf(modifiersAll),
      PropTypes.arrayOf(PropTypes.oneOf(modifiersAll)),
    ]),
    /** Width className */
    widthClass: PropTypes.oneOf(widths),
  }

  static defaultProps = {
    className: undefined,
    modifier: undefined,
    type: 'text',
    widthClass: undefined,
  }

  render() {
    const {
      className,
      modifier,
      type,
      widthClass,
      ...other
    } = this.props;

    const modifierClassNames = Array.isArray(modifier)
      ? modifier.map(eachModifier => `Input--${eachModifier}`)
      : { [`Input--${modifier}`]: modifier };

    return (
      <input
        className={classnames(
          className,
          `Input Input-${capitalizeFirstLetter(type)}`,
          modifierClassNames,
          widthClass,
        )}
        type={type}
        {...other}
      />
    );
  }
}
