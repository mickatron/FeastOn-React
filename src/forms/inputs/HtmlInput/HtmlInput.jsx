import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import classNameType from 'js/shared/proptypes/className';

import capitalizeFirstLetter from 'js/shared/util/capitalizeFirstLetter';

export default class HtmlInput extends React.PureComponent {

  static propTypes = {
    className: classNameType,
    type: PropTypes.oneOf([
      'checkbox',
      'email',
      'file',
      'hidden',
      'image',
      'number',
      'password',
      'radio',
      'range',
      'reset',
      'search',
      'tel',
      'text',
      'time',
      'url',
    ]).isRequired,
  }

  render() {
    const {
      className,
      type,
      ...other
    } = this.props;

    return (
      <input
        className={classnames(className, `c-${capitalizeFirstLetter(type)}Input`)}
        type={type}
        { ...other }
      />
    ); 
  }
}
  