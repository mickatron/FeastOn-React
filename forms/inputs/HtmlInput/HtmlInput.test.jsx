import { shallow } from 'enzyme';
import React from 'react';

import HtmlInput from './HtmlInput';

describe('HtmlInput', () => {
  let component;
  beforeAll(() => {
    component = shallow((
      <HtmlInput
        className="test"
        modifier="test"
        type="text"
      />
    ));
  });

  describe('renders', () => {
    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });
    it('has passed in className', () => {
      expect(component.find('.Input.test').exists()).toBeTruthy();
    });
    it('has passed in modifier', () => {
      expect(component.find('.Input--test').exists()).toBeTruthy();
    });
  });
});
