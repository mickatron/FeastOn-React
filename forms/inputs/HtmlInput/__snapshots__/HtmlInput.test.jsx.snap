// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlInput renders matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HtmlInput
    className="test"
    modifier="test"
    type="text"
    widthClass={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "test Input Input-Text Input--test",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "test Input Input-Text Input--test",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
